preview/Win7Msix/Win7MSIXInstallerLib/inc/IPackageManager.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <functional>
+#include "IPackageInfo.hpp"
+#include "DeploymentOptions.hpp"
+#include "DeploymentResult.hpp"
+
+namespace Win7MsixInstallerLib {
+    class IPackageManager
+    {
+    public:
+        virtual bool AddPackage(const std::wstring & packageFilePath, DeploymentOptions options, std::function<void(const DeploymentResult &)> callback = nullptr) = 0;

```

Comment:
>bool [](start = 16, length = 4)

why not just return the HRESULT?

Reply:
HResult is very C++ oriented, I preferred to create a more generic interface and be closer to Win10 APIs

Reply:
what Win10 APIs are you referring to?

---
In reply to: [270573968](https://github.com/Microsoft/msix-packaging/pull/69#discussion_r270573968) [](ancestors = 270573968)

Reply:
See the first post, the idea is to mimick existing Windows 10 WinRT API: Windows::ApplicationModel::PackageManager, Windows::ApplicationModel::Package, etc... and they never return a HRESULT. 

The error code (with errorMessage) will be added to `DeploymentResult` (I will do it today).